[OSDOCS-11048] IBM Z CEX: Use Butane's boot_device layouts

travier opened the pull request:
modules/ibm-z-nbde: Do not set luks options

This is already set by Butane when translating the Butane config to a
Machine Config.

Also comment AsciiDoc callouts in yaml.

See: https://github.com/coreos/rhel-coreos-config/issues/10
Fixes: c8241d6688 add NBDE encryption for IBM Z

---

modules/ibm-z-cex: Use Butane's boot_device layouts

Use Butane's boot_device layouts to setup the LUKS encryption with CEX
to properly account for all cases (zVM, LPAR, KVM and DASD or FCP disks).

Remove the NVMe case as it is currently not tested and we don't have
Butane sugar for it.

Do not use the sugar for the zFCP case as there is currently an issue
with mulitpath support with it.

Also remove the `--pbkdf pbkdf2` as it is the default in RHEL and
cryptsetup will refuse to create a LUKS device with another PBKDF
function in FIPS mode.

See: https://coreos.github.io/butane/upgrading-openshift/#luks-cex-support
See: https://coreos.github.io/butane/upgrading-openshift/#boot_device-layouts-s390x-support

---

Version(s): 4.19+

Issue: https://issues.redhat.com/browse/OSDOCS-11048

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.

Additional information:
CC @madhu-pillai @SNiemann15

Updates: https://github.com/openshift/openshift-docs/pull/93101

travier commented:
Reading the [LPAR preview](https://93101--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/upi/installing-ibm-z-lpar#configuring-luks-encryption-via-cex-ibm-z-linuxone-environment_installing-ibm-z-lpar) from https://github.com/openshift/openshift-docs/pull/93101, it mentions NVME devices and it looks like we don't have a layout for that in Butane? Should we use the zfcp layout for those as well?

madhu-pillai commented:
> Reading the [LPAR preview](https://93101--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/upi/installing-ibm-z-lpar#configuring-luks-encryption-via-cex-ibm-z-linuxone-environment_installing-ibm-z-lpar) from #93101, it mentions NVME devices and it looks like we don't have a layout for that in Butane? Should we use the zfcp layout for those as well?
> 
> Edit: That won't work as we validate on `/dev/sd*` for the device, so we would need an additional layout or another case in the docs for NVME, without sugar 😞.

There was no task for NVME device CEX test in LPAR. Only zKVM(virt), zVM(FCP and DASD). I am being absent mind to validate that. :😞

 @werled  - Was there any task for NVME on LPAR CEX test. In my jira it was not.

travier committed:
ibm-z: Use Butane's boot_device layouts for CEX

modules/ibm-z-cex: Use Butane's boot_device layouts

Use Butane's boot_device layouts to setup the LUKS encryption with CEX
to properly account for all cases (zVM, LPAR, KVM and DASD or FCP disks).

Remove the NVMe case as it is currently not tested and we don't have
Butane sugar for it.

Do not use the sugar for the zFCP case as there is currently an issue
with mulitpath support with it.

Also comment AsciiDoc callouts in yaml.

See: https://coreos.github.io/butane/upgrading-openshift/#luks-cex-support
See: https://coreos.github.io/butane/upgrading-openshift/#boot_device-layouts-s390x-support

---

modules/ibm-z-nbde & ibm-z-cex: Do not set luks options

This is already set by Butane when translating the Butane config to a
Machine Config.

Also remove the `--pbkdf pbkdf2` as it is the default in RHEL and
cryptsetup will refuse to create a LUKS device with another PBKDF
function in FIPS mode.

See: https://github.com/coreos/rhel-coreos-config/issues/10
Fixes: c8241d6688 add NBDE encryption for IBM Z

travier commented:
The docs also currently don't include the butane commands top convert the configs to MachineConfigs and from memory those commands are there on other pages.

werled commented:
@travier - since NVMe is not explictly tested, I think we should exclude it for now from the documented procedure (and from the note stating "Specifies a unique fully qualified path depending on disk type. This can be DASD-type, FCP-type, or NVMe-type disks."), properly test it and then get this docs change in as soon as we have the verification.

We can decide if we need additional sugar or if we handle it with the currently available config but as far as I understand, in the latter case we would be able to handle this as a docs bug fix as soon as verification is there.

travier commented:
I've also removed the explicit `--pbkdf pbkdf2` option as this is the default in RHEL and setting it explicitly to something else in FIPS mode fails in cryptsetup thus it's not needed.

madhu-pillai commented:
Thank you @travier for the updates in the document. It looks good to me.

travier commented:
Updated to re-add and update the note about the karg.

travier commented:
Updated again to not use the sugar on with zFCP disks as there is apparently an issue when used with multipath.

madhu-pillai commented:
https://github.com/coreos/fedora-coreos-tracker/issues/1958 - Issue raised for zFCP disk with multipath.

madhu-pillai commented:
Thanks again @travier , The changes for zFCP look good to me.

madhu-pillai commented:
hi, @lg-rh In absence @SNiemann15 , could you please have look into this?
CC @werled

werled commented:
/lgtm from OCP QE side, let's wait for @lg-rh's input from doc pov

lg-rh commented:
/lgtm

agantony reviewed (commented):
Highlighted a few style guide issues; overall lgtm!

- Please squash the commits to a single commit.

travier commented:
I don't understand your suggestion to uncomment the callouts in the source blocks. It's contradictory with https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks.

travier commented:
> * Please squash the commits to a single commit.

Why? There are multiple distinct changes, they don't belong in a single commit.

travier commented on modules/ibm-z-configure-hw-based-cex-encryption.adoc:
```diff
@@ -51,91 +52,112 @@ metadata:
+  kernel_arguments:
+    - rd.luks.key=/etc/luks/cex.key <2>
+----
+<1> Specifies whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the cryptography modules that are provided with {op-system} instead.
```
The existing RHEL documentation does not say `the FIPS mode` but `FIPS mode` only: https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/security_hardening/switching-rhel-to-fips-mode_security-hardening

agantony replied:
> > If you do not specify `true` or `false`, `false` is selected automatically.
> 
> This is different from the rest of the docs as well: https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html-single/installing_on_ibm_z_and_ibm_linuxone/index#installing-ibm-z
> 
> It's kind of out scope for this PR as well.

I appreciate hat you are looking at the existing documentation. It is important to maintain doc consistency, but at instances where there is a deviation from the guidelines we need to update it. I agree that the older instances are out of scope, but I think you need to update the new content that you added as part of this PR. 

The guidelines tell us avoid using negative sentence contsrution:

agantony commented:
> > * Please squash the commits to a single commit.
> 
> Why? There are multiple distinct changes, they don't belong in a single commit.

Please refer to https://docs.google.com/document/d/1fLMpK4bqthtFlCwA36yeo2cI-pExFpX76VMnSi_In6Q/edit?pli=1&tab=t.0#bookmark=id.9bn6glcn7tle

agantony commented on modules/ibm-z-configure-hw-based-cex-encryption.adoc:
```diff
@@ -51,91 +52,112 @@ metadata:
-      name: root
-      wipe_volume: true
+ifndef::ibm-z-kvm[]
+For installations on DASD-type disks, create a file named `main-storage.bu` with the following example Butane configuration for a control plane node with disk encryption:
```
https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#handling-nonsequential-actions

travier commented:
> > I don't understand your suggestion to uncomment the callouts in the source blocks. It's contradictory with [`main`/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks).
> 
> If you notice, callout 4 is uncommented in the example

Indeed, I'll fix that.

xenolinux reviewed (commented):
Could you please squash your commits to merge the PR?
